File: basic_roscpp/src/simple_node_sub_obj.cpp
// Copyright 2020 Juan Carlos Manzanares Serrano

#include "ros/ros.h"
#include "std_msgs/Int64.h"

class Suscriptor
{
public:
    Suscriptor()
    {
        sub = n.subscribe("/contador", 1, &Suscriptor::contadorCallback, this);
    }

    void contadorCallback(const std_msgs::Int64::ConstPtr& msg)
    {
        ROS_INFO("Contador: [%ld]", msg->data);
    }

private:
    ros::NodeHandle n;

    ros::Subscriber sub;
};

int main(int argc, char **argv)
{
    ros::init(argc, argv, "simple_node_sub_obj");

    Suscriptor sub;

    ros::spin();

    return 0;
}

File: basic_rospy/src/simple_node_sub_obj.py
#! /usr/bin/env python3

import sys
import rospy
from std_msgs.msg import Int32

class Subscriber():
    def __init__(self):
        self.sub = rospy.Subscriber("/contador", Int32, self.contadorCallback)

    def contadorCallback(self, msg):
        rospy.loginfo("Contador: " + str(msg.data))

def main(args=None):
    rospy.init_node("simple_node_sub_obj")

    sub = Subscriber()

    rate = rospy.Rate(1)

    while(not rospy.is_shutdown()):
        rate.sleep()

    rospy.spin()

if __name__ == "__main__":

    try:
        main(sys.argv)
    except rospy.ROSInterruptException:
        pass